ext/compressors/iaa/iaacodec/include/iaaInterface-c.h

```diff
@@ -0,0 +1,23 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <wiredtiger.h>
+#include <wiredtiger_ext.h>
+
+#ifndef IAA_COMPRESS_C_H
+#define IAA_COMPRESS_C_H
+
+
+#ifdef __cplusplus
```

Comment:
Are the two `#ifdef __cplusplus` correct?
Can we do:
```suggestion
#ifdef __cplusplus
extern "C"
{
#endif
uint32_t getMaxCompressedDataSize(uint32_t srcLen);
uint32_t doCompressData(WT_COMPRESSOR *compressor, WT_SESSION *session, uint8_t *source, uint32_t source_size, uint8_t *dest, uint32_t dest_size);
void doDecompressData(WT_COMPRESSOR *compressor, WT_SESSION *session, uint8_t *source, uint32_t source_size, uint8_t* dest, uint32_t uncompressed_size, uint32_t* result_lenp);
}
#endif

```

Reply:
Sorry, I think you committed my suggestion and it ended up creating duplicate code. I should have probably not used the "suggest" feature, we should probably revert cdf70663da6622a31f851a25dc6629141fe87580.
I was suggesting the following:

```
#ifdef __cplusplus
extern "C" {
...
}
#endif
```

Sorry for the confusion.

Reply:
Hi, etienneptl
Thank you for reply. 
I have one question about using only one __cplusplus:

1. If iaaInterface-c.h is built by c compilation,  doCompressData funtion is not found.  error like:
wiredtiger/ext/compressors/iaa/iaa_compress.c: In function ‘iaa_compress’:
wiredtiger/ext/compressors/iaa/iaa_compress.c:81:27: error: implicit declaration of function ‘doCompressData’ [-Werror=implicit-function-declaration]
     uint32_t result_len = doCompressData(compressor, session, src, (uint32_t)src_len, dst, (uint32_t)dst_len);

Reply:
I am sorry, I had another look and your first proposal (https://github.com/wiredtiger/wiredtiger/pull/8905/commits/428b8a56b9000b00f2d06881ec0330ba16700361#diff-a6f1ab297e90dbcfc10534d8c5c238f124fa16299ff99d604fd15e21e7d00e7a) was totally right, we should get back there.

We only need to fix the pragma once and the extra line before the first `#ifdef __cplusplus`. Sorry again.

Reply:
Got it.